pywinauto/controls/uia_controls.py

```diff
@@ -612,25 +616,20 @@ def column_count(self):
         """Return the number of columns"""
         if self.iface_grid_support:
             return self.iface_grid.CurrentColumnCount
-        elif self.is_table and len(self.children()) > 0:
-            return self.__resolve_col_count(len(self.children()[0].children()))
-        else:
-            # ListBox doesn't have columns
-            return 0
+        elif self.is_table:
+            self.__raise_not_implemented()
+        # ListBox doesn't have columns
+        return 0
 
     # -----------------------------------------------------------
-    def get_column_header_items(self):
-        """Return Headers of each column (top of DataGrid)"""
-        if self.col_header:
-            return get_item(0).children(control_type="Header")
-        return []
+    def get_header_controls(self):
+        """Return Header controls associated with the Table"""
+        try:
+            hdr = self.children(control_type="Header")
```

Comment:
It could be direct `return self.children(control_type="Header")` and `return None` below.